drone/build_info.go

```diff
@@ -32,16 +32,28 @@ func buildInfo(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	number, err := strconv.Atoi(c.Args().Get(1))
-	if err != nil {
-		return err
-	}
+	buildArg := c.Args().Get(1)
 
 	client, err := newClient(c)
 	if err != nil {
 		return err
 	}
 
+	var number int
+	if buildArg == "last" {
+		// Fetch the build number from the last build
+		build, err := client.BuildLast(owner, name, "")
+		if err != nil {
+			return err
+		}
+		number = build.Number
+	} else {
+		number, err = strconv.Atoi(buildArg)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I would recommend altering the logic slightly to only invoke `client.Build` once when `last` is the build argument. The second call is redundant.

```
var build *model.Build
if buildArg == "last" {
  build, err = client.BuildLast(owner, name, "")
  ...
} else {
  number, err = strconv.Atoi(buildArg)
  ...
  build, err = client.Build(owner, name, number)
}
```